src/umpire/strategy/AllocationTracker.cpp

```diff
@@ -42,10 +42,7 @@ AllocationTracker::allocate(size_t bytes)
 void
 AllocationTracker::deallocate(void* ptr)
 {
-  auto record = deregisterAllocation(ptr);
-
-  UMPIRE_CHECK_ALLOCATOR(record, m_allocator->getName());
```

Comment:
Is there a similar check now being done in `deregisterAllocation`?  I do not see this macro being called there.

Reply:
Yes. I checked for leaks and found the bug: deallocating after attempting to deregister with the wrong allocator fails because it was already removed from the allocation map. I fixed it adding `findAllocator` to `Inspector`, but that slowed the code down to almost previous performance. To recover the performance I do the check inside `Inspector::deregisterAllocation(void*)`.